tracing-subscriber/src/layer.rs

```diff
@@ -235,13 +314,32 @@ where
     S: Subscriber,
 {
     fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
-        let interest = self.inner.register_callsite(metadata);
-        self.layer.register_callsite(metadata, interest)
+        let outer = self.layer.register_callsite(metadata);
+        if outer.is_never() {
+            // if the outer layer has disabled the callsite, return now so that
+            // the subscriber doesn't get its hopes up.
+            return outer;
```

Comment:
so does this mean if _one_ layer is never interested in a callsite then the others will never be interested? How would this work with something like metrics where we are only interested in some of the sites but not all while those sites we are not interested in might want to be printed?

Reply:
As I think we discussed earlier, the `enabled` and `register_callsite` methods on `Layer` are intended specifically for implementing filter layers that _should_ effect whether a span/event is enabled globally. If an _individual_ layer is not interested in a given span/event, but it doesn't want to disable that for the rest of the stack, it should *not* disable it from `enabled`/`register_callsite`. 

We could potentially add additional methods to allow a layer to indicate that it is individually not interested in a span/event, but I decided to punt on that for a follow-up PR.

Reply:
I'll make sure the docs make it clear what the intended use-case is for these methods.

Reply:
Improved the docs on filtering in 41f67e712fa632c94725013366473a182b0d544a